app/controllers/posts_controller.rb

```diff
@@ -1,30 +1,40 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.where(user_id: params[:user_id])
+    @posts = Post.includes(:user).where(user_id: params[:user_id]).references(:posts)
     @user = User.find(params[:user_id])
   end
 
   def show
-    @post = Post.find(params[:id])
+    @post = Post.includes(:user).find(params[:id])
+    @comments = @post.comments.includes(:user)
   end
 
   def new
     @post = Post.new
+    render :new
   end
 
   def create
-    @post = Post.new(post_params)
-    @post.user_id = params[:user_id]
-    if @post.save
-      redirect_to user_posts_path
+    post = Post.new(params[:post].permit(:title, :text, :user_id))
+    post.user = current_user
+
+    if post.save
+      flash[:notice] = 'Post was created successfully.'
+      redirect_to user_posts_path(current_user, post)
     else
-      render :new
+      flash.now[:error] = 'Post was not created.'
+      render :show
```

Comment:
- [ ] When I tried to create a post I got the following error please make sure to fix that in order user can create a post without errors
<img width="1434" alt="Screen Shot 2022-10-04 at 17 46 21" src="https://user-images.githubusercontent.com/61790587/193880327-3caea10e-156b-4cfb-928d-2ff02cd963bb.png">

Reply:
Hello @youmari  this is because you have not signed in , after creating a user sign it to create a post

Reply:
@youmari I've edited the code and signed_in and didn't work